examples/stable_diffusion_v2/ldm/data/dataset.py

```diff
@@ -63,15 +65,23 @@ def load_data(
         filter_small_size=filter_small_size,
     )
     datalen = dataset.__len__
+
     loader = build_dataloader_ft(dataset, datalen, t2i_collate, batch_size, device_num, rank_id=rank_id)
```

Comment:
This part of code is DIFF from Luoyang's implementation as I remembered, which is 
``` python
device_num_per_workers = get_local_rank_size()
shard_id = get_local_rank()
loader = build_dataloader_ft(dataset, datalen, t2i_collate, batch_size, device_num_per_workers , rank_id=shard_id )
...
batchlen = datalen // (batch_size * device_num_per_workers )  ## pay attention here
```